flight/search/plot_algo.py

```diff
@@ -0,0 +1,601 @@
+    to return the actual coordinates.
+
+    Methods
+    -------
+    __prep_grid(cell_map: CellMap) -> NDArray
+        turns the CellMap into a numpy array
+    get_valid_point(point: tuple[int, int], cell_map: CellMap, cell_size: int) -> tuple[int, int]
+        given coordinates outside the bounded area, find the nearest point inside the area
+    __decompress_point(point: tuple[int, int], cell_map: CellMap, cell_size: int) -> tuple[int, int]
+        given the coordinates of a compressed point, return the uncompressed coordinates in the
+        center of that point
+    decompress_route(
+        route: list[tuple[int, int]],
+        cell_map: CellMap, cell_size: int
+        ) -> list[tuple[int, int]]
+
+        given a compressed path, return the uncompressed path
+    """
+
+    @staticmethod
+    def __prep_grid(cell_map: CellMap) -> NDArray[int8]:
+        """
+        Creates a numpy array of the given CellMap for faster computation
+
+        Parameters
+        ----------
+        cell_map: CellMap
+            the CellMap to turn into a numpy array
+
+        Returns
+        -------
+        numpy_grid: NDArray
+            the numpy version of the uncompressed CellMap
+        """
+        new_grid: NDArray[int8] = zeros((len(cell_map.data), len(cell_map[0])), dtype=int8)
+        i: int
+        j: int
+
+        for i in range(len(cell_map.data)):
+            for j in range(len(cell_map[0])):
+                if cell_map[i][j].is_valid:
+                    new_grid[i][j] = 1
+        return new_grid
+
+    @staticmethod
+    def get_valid_point(
+        point: tuple[int, int], cell_map: CellMap, cell_size: int
+    ) -> tuple[int, int]:
+        """
+        given coordinates outside the bounded area, find the nearest point inside the area
+
+        Parameters
+        ----------
+        point: tuple[int, int]
+            the point lying outside the bounded area
+        cell_map: CellMap
+            the CellMap being searched
+        cell_size: int
+            the size of the compressed cells
+
+        Returns
+        -------
+        nearest_point: tuple[int, int]
+            the nearest point within the bounded area
+        """
+        cell_map.display()
+        point = (point[0] * cell_size, point[1] * cell_size)
+        middle: int = cell_size // 2
+        closest: float = float("inf")
+        closest_point: tuple[int, int] = (point[0] + middle, point[1] + middle)
+
+        i: int
+        j: int
+        for i in range(cell_size):
+            for j in range(cell_size):
+                try:
+                    if cell_map[point[0] + i][point[1] + j].is_valid:
+                        dist: float = sqrt((middle - i) ** 2 + (middle - j) ** 2)
+                        if dist < closest:
+                            closest = dist
+                            closest_point = (point[0] + i, point[1] + j)
+                except IndexError:
+                    pass
```

Comment:
Should try to constrain i/j rather than throwing and index error. Can be accomplished with an if statement that checks if the index is within the correct range.